packages/blade/src/components/Button/Button/Button.tsx

```diff
@@ -0,0 +1,57 @@
+import type { IconComponent } from '../../Icons';
+import BaseButton from '../BaseButton';
+
+type ButtonCommonProps = {
+  variant?: 'primary' | 'secondary' | 'tertiary';
+  size?: 'large' | 'medium' | 'small' | 'xsmall';
+  iconPosition?: 'left' | 'right';
+  isDisabled?: boolean;
+  isFullWidth?: boolean;
+  onClick?: () => void;
+  type?: 'button' | 'reset' | 'submit';
+};
+
+/*
+  Mandatory children prop when icon is not provided
+  */
+type ButtonWithoutIconProps = ButtonCommonProps & {
+  icon?: undefined;
```

Comment:
check [this](https://www.typescriptlang.org/play?ssl=26&ssc=1&pln=27&pc=1#code/JYWwDg9gTgLgBAJQKYEMDG8BmUIjgIilQ3wG4AocmATzCTgCEBXGGCAOwGFcQOAFHGADOcALxwA3uThwAbiijAU7GAH4AXATCKQC6vjgAfAkKRoOAEz0Hj+GElhKo+ijKHAAXkg0EANgoBzJBsCECQLYCYQEPwhXV9fGIAPOJQEsmk4YHN2Pgh3GGAOH3xfJEwYGMUAgAtK1yyhABFgIRQAIzKLH3aICDLlBtaAMSYEgHVgCxganr6B9gaOTl9sgGsfAAoASjEAPjkIKYaaOhL2ljZ2KqRTSqMTJnaQYHryAF8KKlp6ZlYOSYzCAsACSOQEEGEYkYl2WPH4ghEADJJJlssVNEx2BZysB2OEGmgasBfBYiOxNEIYIp2AEKJ9KKdfrD2ICamCEZCROI-lduCBeLlEXAUVIZOiKXAqTS6ZkiSSyUh2D5pXjZQzvnQYf8hVzobyAa8asCYBzdVDjAbWUazRDhF8kElILA4DkqdqrporXbuaiZPLSeTNPgVus4ABZJAAQnwABoPhQgA) out? Will this work to get rid of `icon?: undefined`?